src/main/java/teammates/logic/api/TaskQueuer.java

```diff
@@ -294,6 +294,21 @@ public void scheduleEmailsForSending(List<EmailWrapper> emails) {
         }
     }
     
+    /**
+     * Schedules production of searchable documents
```

Comment:
On option 1: Task queues _are_ testable as units. The problem only appears in black-box testing (browser-based tests fall into this category) because there's no way to precisely determine when the task queues will finish (in fact, if black-box tests can find a way to circumvent that problem, the "black-box" itself is compromised). The only way is to guess, i.e. using a fixed wait (one minute or so) or querying the back-end repeatedly until the data appears.

On option 2: once again, we have #2258 which is where the real bottleneck is.